### PR TITLE
Update dotnet-sdk from version 2.2.106 to 2.2.204

### PR DESCRIPTION
**Pull request by charlesbalderson**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

### CODE REVIEW

File: Casks/dotnet-sdk.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
 cask 'dotnet-sdk' do
-  version '2.2.106'
-  sha256 'b37c72440be563faca0a6044354c73bff67d55e20ab662d1c56cc9c21d27ffc7'
+  version '2.2.204'
+  sha256 'bed6633d1e1811417cea2d701423ff0dec241f441b42395f618ec5bf5476e89a'
 
-  url "https://download.visualstudio.microsoft.com/download/pr/4d0f3f47-4c25-4102-8df8-a6fe7b472677/d1e3684501ffb70df10879b831b2e70e/dotnet-sdk-#{version}-osx-x64.pkg"
+  url "https://download.visualstudio.microsoft.com/download/pr/fa52799f-b3cd-4d79-9869-a6b3997b96d7/20479fc2b7925ef40592a40ef89f3099/dotnet-sdk-#{version}-osx-x64.pkg"
```
Comment (reitermarkus) [inline Casks/dotnet-sdk.rb]: This should be added to the version.

**Comment by core-code** [comment]
the download page shows '2.2.107' as the newest version?

**Comment by anarsen** [comment]
> the download page shows '2.2.107' as the newest version?

2.2.107 is the latest one compatible with Visual Studio for Mac.

This page shows 2.2.204 to be the latest stable SDK: https://dotnet.microsoft.com/download/dotnet-core/2.2#sdk-2.2.204

**Comment by anarsen** [comment]
@reitermarkus What's missing in this one? I'll try and pick it up.

**Comment by vitorgalvao** [comment]
> @reitermarkus What's missing in this one? I'll try and pick it up.

https://github.com/Homebrew/homebrew-cask/pull/63431#discussion_r285070657

File: Casks/dotnet-sdk.rb (hunk 1 above)
Comment (anarsen) [reply Casks/dotnet-sdk.rb]: @reitermarkus Do you mean the GUID-y components of the URL?

Comment (vitorgalvao) [reply Casks/dotnet-sdk.rb]: > Do you mean the GUID-y components of the URL?

Yes. See [`visual-studio-code`](https://github.com/Homebrew/homebrew-cask/blob/d7a427f7bc01a0fa270e649b033e892ee86a9599/Casks/visual-studio-code.rb).

**Comment by Svetomech** [comment]
Should either be updated to **2.2.107** (at the time of writing this comment) or **2.2.300**, I think.
Created [a PR](https://github.com/Homebrew/homebrew-cask/pull/63905) for **2.2.107** just in case. 

Maybe introduce a new Cask for versions not compatible with Visual Studio for Mac? ~~Even though I personally don't use it, many still do~~.

**Comment by vitorgalvao** [comment]
Closing in favour of https://github.com/Homebrew/homebrew-cask/pull/63905.

**Pull request closed without merging**